torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -615,20 +630,98 @@ NVFUSER_DEFINE_BINARY_FLOAT_OP(atan2, Atan2)
+        "input must have integral or boolean type, but got ",               \
+        v1->dtype(),                                                        \
+        " and ",                                                            \
+        v2->dtype());                                                       \
+    return binaryOp(                                                        \
+        BinaryOpType::op_type, v1, v2, TypePromotion::default_op_config);   \
+  }                                                                         \
+  TensorView* op_name(TensorView* v1, TensorView* v2) {                     \
+    TORCH_CHECK(                                                            \
+        (isIntegralType(v1->dtype()) || v1->dtype() == DataType::Bool) &&   \
+            (isIntegralType(v2->dtype()) || v2->dtype() == DataType::Bool), \
+        "input must have integral or boolean type, but got ",               \
+        v1->dtype(),                                                        \
+        " and ",                                                            \
+        v2->dtype());                                                       \
+    return binaryOp(                                                        \
+        BinaryOpType::op_type, v1, v2, TypePromotion::default_op_config);   \
+  }
+
+NVFUSER_DEFINE_BITWISE_OP(bitwise_and, And)
+NVFUSER_DEFINE_BITWISE_OP(bitwise_or, Or)
+NVFUSER_DEFINE_BITWISE_OP(bitwise_xor, Xor)
+#undef NVFUSER_DEFINE_BITWISE_OP
+
+#define NVFUSER_DEFINE_BITWISE_SHIFT_OP(op_name, op_type)        \
+  Val* op_name(Val* v1, Val* v2) {                               \
+    TORCH_CHECK(                                                 \
+        isIntegralType(v1->dtype()),                             \
+        "input must have integral or boolean type, but got ",    \
+        v1->dtype());                                            \
+    return binaryOp(BinaryOpType::op_type, v1, v2, v1->dtype()); \
+  }                                                              \
+  TensorView* op_name(TensorView* v1, Val* v2) {                 \
+    TORCH_CHECK(                                                 \
+        isIntegralType(v1->dtype()),                             \
+        "input must have integral or boolean type, but got ",    \
+        v1->dtype());                                            \
+    return binaryOp(BinaryOpType::op_type, v1, v2, v1->dtype()); \
+  }                                                              \
+  TensorView* op_name(Val* v1, TensorView* v2) {                 \
+    TORCH_CHECK(                                                 \
+        isIntegralType(v1->dtype()),                             \
+        "input must have integral or boolean type, but got ",    \
+        v1->dtype());                                            \
+    return binaryOp(BinaryOpType::op_type, v1, v2, v1->dtype()); \
```

Comment:
that's not strictly true, as eager mode promotes between dtypes (so e.g v2 being int64 will cause result to be int64), but that's a corner case.